autogen/types.py

```diff
@@ -0,0 +1,17 @@
+from typing import Dict, Literal, TypedDict
+
+
+class UserMessageTextContentPart(TypedDict):
```

Comment:
Nice design. It will be compatible with C# and .NET implementations. Moreover, it can help us migrate away from OpenAI's formatting style.

However, we will need much more detailed documentation here to ensure that users are not confused.

Reply:
For the moment, this is formalizing in types what is already the case implicitly. I do agree that for it to make sense it'll need to be documented